src/Models/Nonhydrostatic2DModels/equations_pressure.jl

```diff
@@ -21,3 +23,29 @@ end
 
     return p
 end
+
+@inline function calculate_pressure(
+    Y,
+    Ya,
+    ::TotalEnergy,
+    ::EquilibriumMoisture,
+    params,
+    Φ,
+    K,
+    FT,
+)
```

Comment:
Maybe this should be `calculate_pressure!` and be non-allocating?

Reply:
hmm, do you mean put `p` in cache and cache is an input arg for this function?